test/matchAnyPCat.js

```diff
@@ -0,0 +1,154 @@
+                    {
+                        "id": 'b',
+                        "cat": 'w',
+                    },
+                    {
+                        "id": 'c',
+                        "cat": 'w',
+                    }
+                ]
+            }
+        ]
+    }; 
+
+    var ptree2 = {
+        "id": 'root',
+        "cat": 'i',
+        "children": [
+            {
+                "id": 'c',
+                "cat": 'w',
+            },
+            {
+                "id": 'cp',
+                "cat": 'phi',
+                "children": [
+                    {
+                        "id": 'a',
+                        "cat": 'w',
+                    },
+                    {
+                        "id": 'b',
+                        "cat": 'w',
+                    }
+                ]
+            }
+        ]
+    }; 
+        
+    var stree2 = {
+        "id": "CP1",
+        "cat": "cp",
+        "children": [ 
+            {
+                "cat": "xp", 
+                "id": "XP_8", 
+                "children": [ 
+                    {
+                        "id": "a", 
+                        "cat": "x0"
+                    } 
+                ] 
+            },
+            { 
+                "cat": "xp", 
+                "id": "XP_9", 
+                "children": [ 
+                    {
+                        "id": "b", 
+                        "cat": "x0"
+                    } 
+                ] 
+            },
+            {
+                "cat": "xp", 
+                "id": "XP_10", 
+                "children": [ 
+                    {
+                        "id": "c", 
+                        "cat": "x0"
+                    } 
+                ] 
+            } 
+        ]
+    };
+
+    var ptree3 = {
+        "id": 'root',
+        "cat": 'i',
+        "children": [
+            {
+                "id": 'xp',
+                "cat": 'phi',
+                "children": [
+                    {
+                        "id": 'x',
+                        "cat": 'w'
+                    },
+                    {
+                        "id": 'y',
+                        "cat": 'w'
+                    },
+                    {
+                        "id": 'z',
+                        "cat": 'w'
+                    }
+                ]
+            }
+        ]
+    }; 
+
+    describe("matchAnyPCat.html", function() {
+        describe('Tableau 1: matchSP-cp', function() {
+            it('matchSP-cp', function() {
+                assert.equal(matchSP(stree1, ptree1, 'cp', {anyPCat:true}), 0);
+                assert.equal(matchSP(stree1, ptree2, 'cp', {anyPCat:true}), 1);
+            });
+        });
+        describe('Tableau 2: matchSP-xp', function() {
+            it('matchSP-xp', function() {
+                assert.equal(matchSP(stree2, ptree1, 'xp', {anyPCat:true}), 0);
+                assert.equal(matchSP(stree2, ptree3, 'xp', {anyPCat:true}), 4);
```

Comment:
This particular test is failing right now.

Reply:
Looks like the expected value is actually wrong. 
stree2 = {[a] [b] [c]} --> three XPs to match to anyPCat -> maximum of three possible violations when parameterized to XP.
ptree3 = {(x y z)} --> doesn't match any XP since all ids are different

It would be good to also test:
matchSP(stree2, ptree3, 'any', {anyPCat:true})
This should yield more violations:
1 for the CP
3 for the XPs
3 for the actual terminals (X0s)
= 7 violations, I think